server/pairing/connection.go

```diff
@@ -178,7 +180,7 @@ func (cp *ConnectionParams) validateAESKey() error {
 
 func (cp *ConnectionParams) validateServerMode() error {
 	switch cp.serverMode {
-	case Receiving, Sending:
+	case 0, Receiving, Sending:
```

Comment:
The refactored code doesn't use `ServerMode` anymore, the only place it is used is in the generation and parsing of `ConnectionString`s. So we want to ignore any 0 value `ServerModes` because it doesn't matter.

I've also added a `TODO` to remove `ServerMode` entirely from the process flow, it is no longer needed.